app/javascript/chat/components/ChannelButton.jsx

```diff
@@ -0,0 +1,123 @@
+import { h, Component, createRef } from 'preact';
+import PropTypes from 'prop-types';
+
+class ChannelButton extends Component {
+  buttonRef = createRef();
+
+  componentDidMount() {
+    const { isActiveChannel = false } = this.props;
+
+    if (isActiveChannel) {
+      this.buttonRef.current.click();
+    }
+  }
+
+  renderChannelImage = () => {
+    const { channel, newMessagesIndicator, discoverableChannel } = this.props;
+
+    return (
+      <span
+        data-channel-slug={channel.channel_modified_slug}
+        className={
+          discoverableChannel
+            ? 'chatchanneltabindicator'
+            : `chatchanneltabindicator chatchanneltabindicator--${newMessagesIndicator}`
+        }
+        data-channel-id={channel.chat_channel_id}
+      >
+        <img
+          src={channel.channel_image}
+          alt="pic"
+          className={
+            channel.channel_type === 'direct'
+              ? 'chatchanneltabindicatordirectimage'
+              : 'chatchanneltabindicatordirectimage invert-channel-image'
+          }
+        />
+      </span>
+    );
+  };
+
+  render() {
+    const {
+      channel,
+      handleSwitchChannel,
+      otherClassname,
+      newMessagesIndicator,
+      isUnopened,
+      discoverableChannel,
+      triggerActiveContent,
+    } = this.props;
+
+    return (
+      <button
+        type="button"
+        ref={this.buttonRef}
+        key={channel.id}
+        className="chatchanneltabbutton"
+        onClick={
+          discoverableChannel ? triggerActiveContent : handleSwitchChannel
+        }
+        data-content="sidecar-channel-request"
+        data-channel-id={channel.chat_channel_id}
+        data-channel-slug={channel.channel_modified_slug}
+        data-channel-status={channel.status}
+        data-channel-name={channel.channel_name}
+      >
+        <span
```

Comment:
refactor (non-blocking): Is this span necessary? Can all the data attributes and styling be on the button?